src/realm/object-store/sync/impl/sync_metadata.cpp

```diff
@@ -157,31 +157,6 @@ SyncMetadataManager::SyncMetadataManager(std::string path, bool should_encrypt,
         config.encryption_key = std::move(*encryption_key);
     }
 
-    config.migration_function = [](SharedRealm old_realm, SharedRealm realm, Schema&) {
```

Comment:
This has been dead code for a very long time and we can't actually migrate from the versions which wrote metadata realms with the older schema versions anyway.